packages/node_modules/@ciscospark/react-component-utils/src/adaptiveCardUtil.js

```diff
@@ -49,13 +49,35 @@ export function getCardConfig(card, sparkInstance) {
  * @param {object} addChildNode
  * @returns {object}
  */
-export function getAdaptiveCard(cards, displayName, sparkInstance, addChildNode = null) {
+
+export function getAdaptiveCard(cards, displayName, sparkInstance, addChildNode = null, parentActivity, convo) {
   // Parse the card for rendering
   try {
     const cardConfig = getCardConfig(cards[0]);
     const PatchedAdaptiveCard = getPatchedAdaptiveCard(addChildNode);
     const adaptiveCard = new PatchedAdaptiveCard.AdaptiveCard();
 
+    adaptiveCard.onExecuteAction = (action) => {
+      switch (action.getJsonTypeName()) {
+        case OPEN_URL_ACTION:
+          window.open(action.url);
+          break;
+        case SUBMIT_ACTION:
+          sparkInstance.internal.conversation
```

Comment:
Actions performed on the SDK should be done in the redux action and not within the utilities. This should be an action within the `redux-module-conversation` as well as a reducer state added to show the action being performed. Take a look at the redux styleguide for more info: https://github.com/webex/web-styleguide/tree/master/react/redux